### PR TITLE
BF8 support

### PR DESCRIPTION
**Pull request by rolfmorel**

### CODE REVIEW

File: third_party/cpu/lib/Xsmm/XsmmUtils.cpp [hunk 1]
```diff
@@ -1009,6 +1011,11 @@ buildBrgemmCalls(PatternRewriter &rewriter, Operation *op, ValueRange inputs,
 
   auto dtype = xsmm::utils::getDataType(rewriter, inputs[0].getType());
   auto outDtype = xsmm::utils::getDataType(rewriter, inputs[2].getType());
+  if (dtype.getValue() == DataType::BF8) {
+    //FIXME: Setting either of the following causes a segfault
```
Comment (rolfmorel) [inline third_party/cpu/lib/Xsmm/XsmmUtils.cpp:1015]: This is the major issue.

File: cmake/xsmm.cmake [hunk 2]
```diff
@@ -13,8 +13,8 @@ else()
 
   FetchContent_Declare(
     xsmm
-    URL https://github.com/libxsmm/libxsmm/archive/6a286ec2884abe8e05587dfead1784ddc8cd5390.tar.gz
-    URL_HASH SHA256=0e726aa962faaa3050e3577269a3f5d85434707aa9183c29421b97596fe85b5e
+    URL https://github.com/libxsmm/libxsmm/archive/ed70ae64e0ed25b6d5d42a5d62ffafd9fe0d9b42.tar.gz
+    #URL_HASH SHA256=0e726aa962faaa3050e3577269a3f5d85434707aa9183c29421b97596fe85b5e
```
Comment (rengolin) [inline cmake/xsmm.cmake:17]: not a good idea disabling this

File: third_party/cpu/lib/Xsmm/XsmmUtils.cpp (hunk 1 above)
Comment (rolfmorel) [reply third_party/cpu/lib/Xsmm/XsmmUtils.cpp]: Per discussion with Evangelos, no need for VNNI flags.

File: cmake/xsmm.cmake (hunk 2 above)
Comment (rolfmorel) [reply cmake/xsmm.cmake]: Noted, though in the end didn't need to update libxsmm.

File: third_party/cpu/lib/Xsmm/runtime/XsmmRunnerUtils.cpp [hunk 3]
```diff
@@ -343,11 +344,19 @@ extern "C" int64_t xsmm_brgemm_dispatch(const libxsmm_datatype dtype,
   l_shape.a_in_type = dtype;
   l_shape.b_in_type = dtype;
   l_shape.out_type = out_dtype;
-  // Retarget computation type from bf16 to f32 due to missing hardware support.
-  l_shape.comp_type =
-      out_dtype == LIBXSMM_DATATYPE_BF16 ? LIBXSMM_DATATYPE_F32 : out_dtype;
+  l_shape.comp_type = LIBXSMM_DATATYPE_F32;
```
[the remark this answers is not in the document]
Comment (rolfmorel) [reply third_party/cpu/lib/Xsmm/runtime/XsmmRunnerUtils.cpp:359]: Outdated.

**Comment by rolfmorel** [comment]
@alheinecke  @egeor @rengolin This should now be ready to go.

**Review by rengolin — COMMENTED** [review]
LGTM, but I'll let Alex & Evangelos review before approval.

File: python/tutorials/03-matrix-multiplication-cpu.py [hunk 4]
```diff
@@ -411,14 +411,14 @@ def matmul(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, M: int, N: int, K:
 print(f"triton_cpu_output_with_{a.dtype}_inputs={triton_output}")
 print(f"torch_cpu_output_with_{a.dtype}_inputs={torch_output}")
 rtol = 0
-if torch.allclose(triton_output, torch_output, atol=1e-2, rtol=rtol):
+if torch.allclose(triton_output.type(torch.float64), torch_output.type(torch.float64), atol=1e-2, rtol=rtol):
```
Comment (alheinecke) [inline python/tutorials/03-matrix-multiplication-cpu.py:414]: we could also use float32, or is there a special need for float64.. it's just the "more true" solution.

**Review by alheinecke — APPROVED** [review]
LGMT, minor question

**Review by egeor — APPROVED** [review]
LGTM

File: python/tutorials/03-matrix-multiplication-cpu.py (hunk 4 above)
Comment (rolfmorel) [reply python/tutorials/03-matrix-multiplication-cpu.py:414]: The only reason for the upcasts is that certain basic operations don't work on bf8 in PyTorch. That is,

```
Traceback (most recent call last):
  File "/data/nfs_home/rmorel/dev/triton-cpu.git/raise/python/tutorials/03-matrix-multiplication-cpu.py", line 414, in <module>
    if torch.allclose(triton_output, torch_output, atol=1e-2, rtol=rtol):
RuntimeError: "mul_cpu_reduced_float" not implemented for 'Float8_e5m2'
```

As the widest type we use is f32, I could have used f32 indeed.

**Pull request merged**